pallets/circuit/src/lib.rs

```diff
@@ -1693,47 +1641,78 @@ impl<T: Config> Pallet<T> {
-                            full_side_effect.confirmed = Some(confirmation.clone());
-                            Ok(full_side_effect.clone())
-                        } else {
-                            Err("Attempt to confirm side effect from the next step, \
-                                    but there still is at least one unfinished step")
-                        }
+            // Double check there are some side effects for that Xtx - should have been checked at API level tho already
+            if step_side_effects.is_empty() {
+                return Err("Xtx has an empty single step.")
+            }
+
+            // Find sfx object index in the current step
+            match step_side_effects
+                .iter()
+                .position(|sfx| sfx.input.generate_id::<SystemHashing<T>>() == input_side_effect_id)
+            {
+                Some(index) => {
+                    // side effect found in current step
+                    if step_side_effects[index].confirmed.is_none() {
+                        // side effect unconfirmed currently
+                        step_side_effects[index].confirmed = Some(confirmation.clone());
+                        Ok(step_side_effects[index].clone())
+                    } else {
+                        Err("Side Effect already confirmed")
                     }
-                }
+                },
+                None => Err("Unable to find matching Side Effect in given Xtx to confirm"),
             }
-            Err("Unable to find matching Side Effect in given Xtx to confirm")
         }
 
-        let fsx = confirm_order::<T>(side_effect, confirmation, &mut local_ctx.full_side_effects)?;
-        confirm_inclusion(&fsx)?;
-        confirm_execution(
-            <T as Config>::Xdns::best_available(side_effect.target)?.gateway_vendor,
-            <T as Config>::Xdns::get_gateway_value_unsigned_type_unsafe(&side_effect.target),
+        let mut side_effect_id: [u8; 4] = [0, 0, 0, 0];
+        side_effect_id.copy_from_slice(&side_effect.encoded_action[0..4]);
+
+        // confirm order of current season, by passing the side_effects of it to confirm order.
+        let fsx = confirm_order::<T>(
+            side_effect,
+            confirmation,
+            &mut local_ctx.full_side_effects[local_ctx.xtx.steps_cnt.0 as usize],
+        )?;
+        log::info!("Order confirmed!");
+        // confirm the payload is included in the specified block, and return the SideEffect params as defined in XDNS.
+        // this could be multiple events!
+        let (params, source) = <T as Config>::Portal::confirm_and_decode_payload_params(
+            side_effect.target,
+            fsx.submission_target_height,
+            confirmation.inclusion_data.clone(),
+            side_effect_id.clone(),
+        )
+        .map_err(|_| "SideEffect confirmation failed!")?;
+        // ToDo handle misbehaviour
+        log::info!("Params: {:?}", params);
```

Comment:
```suggestion
        log::debug!("Params: {:?}", params);
```

Please demote any debug messages lower, down to debug or ideally trace, try to look anywhere info is being logged and demote them if you can

Reply:
Will keep that in mind. I'll pick a new level, because debug is super noisy, and I rely on these when something doesn't work